var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -241,20 +267,16 @@ def cmake_args(self):
             'GMX_USE_RDTSCP', str(target.family) in ('x86_64', 'x86')
         ))
 
-        if '+mdrun_only' in self.spec:
-            options.append('-DGMX_BUILD_MDRUN_ONLY:BOOL=ON')
-        else:
-            options.append('-DGMX_BUILD_MDRUN_ONLY:BOOL=OFF')
+        if self.spec.satisfies('@:2020'):
+            options.append(self.define_from_variant('GMX_BUILD_MDRUN_ONLY', 'mdrun_only'))
 
         if '~openmp' in self.spec:
             options.append('-DGMX_OPENMP:BOOL=OFF')
         else:
             options.append('-DGMX_OPENMP:BOOL=ON')
```

Comment:
maybe the same here? :)

Reply:
That's getting beyond the intended scope of the pull request, but it seems appropriate. What is the right way to negate the option? Or, rather, are we guaranteed that the variant choice is present in `self.spec` if not specified by the user? If unspecified by the user, we want to accept the CMake default (which is currently ON if available). To follow the current logic in the package.py, we want to explicitly disable GMX_OPENMP if and only if `~openmp` is requested.

Reply:
The variant always has a value, either chosen by the user or set by the concretizer (which might use the default value or not, depending on constraints). I would not rely on cmake defaults in this way:

```python
if '~openmp' in self.spec:
  options.append('-DGMX_OPENMP:BOOL=OFF')
else:
  # do nothing because cmake's default is ON
```
because if the cmake default ever changes to `Off`, you might end up in a situation where `GMX_OPENMP=On` is never passed to cmake when you set `+openmp`. It's better to explicitly set all variables and to copy the cmake default values to the corresponding variant default values.